source/common/router/config_impl.h

```diff
@@ -127,8 +127,12 @@ class RateLimitPolicyImpl : public RateLimitPolicy {
   // Router::RateLimitPolicy
   bool doGlobalLimiting() const override { return do_global_limiting_; }
 
+  // Router::RateLimitPolicy
+  const std::string& rateLimitKey() const override { return rate_limit_key_; }
+
 private:
   bool do_global_limiting_{};
+  std::string rate_limit_key_;
```

Comment:
I added the ability for getObject() to "allow_empty". If you could clean up the constructor to use initializer lists that would be great, that you can make both of these members const.